src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -344,133 +344,312 @@ public class SqlMutationEngine : IMutationEngine
                     {
-                        string primaryKeyRoute = ConstructPrimaryKeyRoute(context, resultRow);
-                        // location will be updated in rest controller where httpcontext is available
-                        return new CreatedResult(location: primaryKeyRoute, OkMutationResponse(resultRow).Value);
+                        isUpdateResultSet = !PostgresQueryBuilder.IsInsert(resultRow);
                     }
 
-                    // Valid REST updates return OkObjectResult
-                    return OkMutationResponse(resultRow);
-                }
-            }
-            else
-            {
-                DbResultSetRow? mutationResultRow = null;
+                    // When read permissions is configured without database policy, a subsequent select query will not be executed.
+                    // However, the read action could have include and exclude fields configured. To honor that configuration setup,
+                    // any additional fields that are present in the response are removed.
+                    if (isReadPermissionConfiguredForRole && !isDatabasePolicyDefinedForReadAction)
+                    {
+                        IEnumerable<string> allowedExposedColumns = _authorizationResolver.GetAllowedExposedColumns(context.EntityName, roleName, EntityActionOperation.Read);
+                        foreach (string columnInResponse in resultRow.Keys)
+                        {
+                            if (!allowedExposedColumns.Contains(columnInResponse))
+                            {
+                                resultRow.Remove(columnInResponse);
+                            }
+                        }
+                    }
 
-                try
-                {
-                    // Creating an implicit transaction
-                    using (TransactionScope transactionScope = ConstructTransactionScopeBasedOnDbType())
+                    // When the upsert operation results in the creation of a new record, an HTTP 201 CreatedResult response is returned.
+                    if (!isUpdateResultSet)
                     {
-                        mutationResultRow =
-                                await PerformMutationOperation(
-                                    context.EntityName,
-                                    context.OperationType,
-                                    parameters);
-                        transactionScope.Complete();
+                        // Location Header is made up of the Base URL of the request and the primary key of the item created.
+                        // However, for PATCH and PUT requests, the primary key would be present in the request URL. For POST request, however, the primary key
+                        // would not be available in the URL and needs to be appened. Since, this is a PUT or PATCH request that has resulted in the creation of
```

Comment:
```suggestion
                        // would not be available in the URL and needs to be appended. Since, this is a PUT or PATCH request that has resulted in the creation of
```